core/src/main/java/org/jruby/RubyArray.java

```diff
@@ -189,11 +201,15 @@ public static final RubyArray newArrayLight(final Ruby runtime) {
     }
 
     public static RubyArray newArray(Ruby runtime, IRubyObject obj) {
-        return new RubyArray(runtime, new IRubyObject[] { obj });
+        return new RubyArrayOneObject(runtime, obj);
     }
 
     public static RubyArray newArrayLight(Ruby runtime, IRubyObject obj) {
-        return new RubyArray(runtime, new IRubyObject[] { obj }, false);
+        return new RubyArrayOneObject(runtime, obj);
```

Comment:
so it might show up in ObjectSpace now ... thus not really "light" (not sure it matters at all)